### PR TITLE
implemented regular images in finitely bicomplete categories

### PR DESCRIPTION
**Pull request by mohamed-barakat**

### CODE REVIEW

File: gap/ToposDerivedMethods.gi [hunk 1]
```diff
@@ -4,6 +4,19 @@
 # Implementations
 #
 
+## Page 20 in Peter Freyd, Aspect of topoi, Bull. Austral. Math. Soc, 7 (1972)
+AddDerivationToCAP( ImageEmbedding,
```
Comment (zickgraf) [inline gap/ToposDerivedMethods.gi:8]: This must be a final derivation, otherwise things might get inconsistent if `ImageObject` is already installed.

Comment (mohamed-barakat) [reply gap/ToposDerivedMethods.gi:8]: You are right. Shall we move it to CAP and correct it there?

Comment (zickgraf) [reply gap/ToposDerivedMethods.gi:8]: I suggest correcting it here to not block on a review/discussion in CAP.

Comment (mohamed-barakat) [reply gap/ToposDerivedMethods.gi:8]: I will correct it in `CategoricalTowers/Toposes`.

**Pull request merged**